arrays.dd

```diff
@@ -249,9 +252,12 @@ $(H3 $(LNAME2 array-setting, Array Setting))
 
 ---------
 int[3] s;
+int[] d = new int[3];
 int* p;
 
+d[] = 3;     // same as d[0] = 3, d[1] = 3, d[2] = 3
 s[] = 3;     // same as s[0] = 3, s[1] = 3, s[2] = 3
+s = 3;       // also works (because s is a static array)
```

Comment:
That's been heavily discussed, and it will eventually get deprecated. Don't encourage anyone to use static array assignment.
I'm not sure what you mean by "dereference d". Any slice contains of a `ptr` to the data and a `length`. If you do `new int[3]`, then the data will live on the GC heap, while the `ptr` and `length` live inside the slice (it's 16-byte on x64).